### PR TITLE
Feature/Meraki switch module per version 

### PR DESCRIPTION
**Pull request by abprasa**
# Description
The existing code has a generic Meraki module, this new code updates the existing Meraki Module, adds more modules based on the currently available releases, and maps them to the relevant versions and the features they support.

# Impacts
Making it more suitable for users to choose modules based on the version of the switch. These new modules will only have the supported features enabled

# NEW Package(s) required
No additional packages are needed. 

# Feature
The idea of updating the existing list of Meraki Modules defined based on the model number to respective versions is to make it more convenient for users to choose modules based on the version of the switch.

# Delete branch after the merge
YES 

# Checklist
- [ ] Document the feature In the Network Device Config Guide
- [ ] Migration Script
- [ ] Test the new modules on UI

# NEWS file entries
## New Features
* Updating existing switch model for the next release

### CODE REVIEW

**Comment by fdurand** [comment]
It miss the reference of the Meraki::MS in web/constanp.pm

```
 Readonly::Scalar our $EXT_URL_MERAKI_MS             => '^/Meraki::MS';
```

**Comment by abprasa** [comment]
> It miss the reference of the Meraki::MS in web/constant.pm
> 
> ```
>  Readonly::Scalar our $EXT_URL_MERAKI_MS             => '^/Meraki::MS';
> ```

Updated the web/constant.pm for Mearki MS module.

File: addons/upgrade/to-13.2-convert-merakiswitch-types.pl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+print "Nothing to be done\n";
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2023 Inverse inc.
```
Comment (fdurand) [inline addons/upgrade/to-13.2-convert-merakiswitch-types.pl]: 2024

File: addons/upgrade/to-13.2-convert-merakiswitch-types.pl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+use strict;
+use warnings;
+use lib qw(/usr/local/pf/lib);
+use lib qw(/usr/local/pf/lib_perl/lib/perl5);
+use pf::IniFiles;
+use pf::file_paths qw(
+    $switches_config_file
+);
+use File::Copy;
```
Comment (fdurand) [inline addons/upgrade/to-13.2-convert-merakiswitch-types.pl]: remove

File: docs/network/networkdevice/meraki.asciidoc [hunk 3]
```diff
@@ -87,7 +87,13 @@ image::meraki-dpsk-connection-profile-2.jpg[scaledwidth="100%",alt="Connection P
 
 image::meraki-dpsk-connection-profile-3.jpg[scaledwidth="100%",alt="Connection Profile configuration"]
 
-===== Switch MS220-8
+===== MS Switch Modules
+
+MS Modules
+This is a standared switch module designed to support all the legacy Meraki Switch series using MS version earlier than v15.x.
```
Comment (fdurand) [inline docs/network/networkdevice/meraki.asciidoc]: standared -> standard

File: docs/network/networkdevice/meraki.asciidoc [hunk 4]
```diff
@@ -87,7 +87,13 @@ image::meraki-dpsk-connection-profile-2.jpg[scaledwidth="100%",alt="Connection P
...
+
+Meraki MS v15 
+MS v15  module is designed for the switch series operating on MS OS v15. Furthermore, it inherits all functionalities from the old "MS220_8" module, making all the configurations below applicable to MS v15.
```
Comment (fdurand) [inline docs/network/networkdevice/meraki.asciidoc]: extra space to remove

File: lib/pf/Switch/Meraki/MS.pm [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+use base ('pf::Switch::Meraki');
+
+use pf::config qw(
+    $WIRED_802_1X
+    $WIRED_MAC_AUTH
+);
+use pf::constants;
+use pf::util;
+use pf::node;
+use Try::Tiny;
```
Comment (fdurand) [inline lib/pf/Switch/Meraki/MS.pm]: remove

File: lib/pf/Switch/Meraki/MS.pm [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    my ( $self, $radius_request ) = @_;
+    my $client_mac      = ref($radius_request->{'Calling-Station-Id'}) eq 'ARRAY'
+                           ? clean_mac($radius_request->{'Calling-Station-Id'}[0])
+                           : clean_mac($radius_request->{'Calling-Station-Id'});
+    my $user_name       = $self->parseRequestUsername($radius_request);
+    my $nas_port_type   = $radius_request->{'NAS-Port-Type'};
+    my $port            = $radius_request->{'NAS-Port'};
+    my $eap_type        = ( exists($radius_request->{'EAP-Type'}) ? $radius_request->{'EAP-Type'} : 0 );
+    my $nas_port_id     = ( defined($radius_request->{'NAS-Port-Id'}) ? $radius_request->{'NAS-Port-Id'} : undef );
+    my $session_id = $self->getCiscoAvPairAttribute($radius_request, "audit-session-id");
```
Comment (fdurand) [inline lib/pf/Switch/Meraki/MS.pm]: place = like the other one

**Review by fdurand — CHANGES_REQUESTED** [review]

**Pull request merged**